### ISSUE TITLE
Rule proposal: `id-match`

### ISSUE THREAD

**Issue by gajus** [sindresorhus/eslint-plugin-unicorn#1545]
When working with external libraries, it is often undesirable to rename their exports for the sole purpose of conforming with `id-match` rule.

This has been proposed as addition to ESLint in https://github.com/eslint/eslint/issues/14005, but ultimately it was rejected:

> I think this falls under our policies of not adding new options to stylistic rules, so in my opinion the best course of action would be to make a custom rule/plugin.

This rule would be equivalent to `id-match` with an additional option `ignoreImports` that would ignore named imports.

## Fail

Assuming that rule does not permit IDs with `$`:

```js
const ContraImageFragment$key = '';
```

## Pass

Assuming that this rule is configured to ignore imports:

```js
import type { ContraImageFragment$key } from '@/__generated__/ContraImageFragment.graphql';
```

Happy to contribute to the codebase.

**Comment by sindresorhus** [comment]
> Do we have examples of other rules that extend ESLint core rule?

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/4435a10f4c47e8d42769c37d275c3c6f87e61dde/rules/expiring-todo-comments.js#L5

And TypeScript ESLint has many examples of this.

**Comment by sindresorhus** [comment]
> Before I spend too much time on this, is extending from a builtin rule a requirement?

You need to wait until the rule is actually accepted.

**Comment by gajus** [comment]
> You need to wait until the rule is actually accepted.

Worst case scenario, you can not accept the rule... I still need the rule.

**Comment by gajus** [comment]
@sindresorhus What's the evaluation process? i.e. what are we waiting here for?

**Comment by fisker** [comment]
I don't have any patterns in my mind to forbid, but I guess we can accept this, maybe it's easier than use the [no-restricted-syntax](https://eslint.org/docs/rules/no-restricted-syntax) rule.
